x-pack/plugins/lens/public/app_plugin/mounter.tsx

```diff
@@ -11,83 +11,124 @@ import { HashRouter, Route, RouteComponentProps, Switch } from 'react-router-dom
 import { render, unmountComponentAtNode } from 'react-dom';
 import { i18n } from '@kbn/i18n';
 
+import { DashboardFeatureFlagConfig } from 'src/plugins/dashboard/public';
 import { Storage } from '../../../../../src/plugins/kibana_utils/public';
 
 import { LensReportManager, setReportManager, trackUiEvent } from '../lens_ui_telemetry';
 
 import { App } from './app';
 import { EditorFrameStart } from '../types';
 import { addHelpMenuToAppChrome } from '../help_menu_util';
-import { SavedObjectIndexStore } from '../persistence';
 import { LensPluginStartDependencies } from '../plugin';
-import { LENS_EMBEDDABLE_TYPE } from '../../common';
+import { LENS_EMBEDDABLE_TYPE, LENS_EDIT_BY_VALUE } from '../../common';
+import {
+  LensEmbeddableInput,
+  LensByReferenceInput,
+  LensByValueInput,
+} from '../editor_frame_service/embeddable/embeddable';
+import { LensAttributeService } from '../lens_attribute_service';
+import { LensAppServices, RedirectToOriginProps } from './types';
+import { KibanaContextProvider } from '../../../../../src/plugins/kibana_react/public';
 
 export async function mountApp(
   core: CoreSetup<LensPluginStartDependencies, void>,
   params: AppMountParameters,
-  createEditorFrame: EditorFrameStart['createInstance']
+  mountProps: {
+    createEditorFrame: EditorFrameStart['createInstance'];
+    getByValueFeatureFlag: () => Promise<DashboardFeatureFlagConfig>;
+    attributeService: LensAttributeService;
+  }
 ) {
+  const { createEditorFrame, getByValueFeatureFlag, attributeService } = mountProps;
   const [coreStart, startDependencies] = await core.getStartServices();
-  const { data: dataStart, navigation, embeddable } = startDependencies;
-  const savedObjectsClient = coreStart.savedObjects.client;
-  addHelpMenuToAppChrome(coreStart.chrome, coreStart.docLinks);
+  const { data, navigation, embeddable } = startDependencies;
+
+  const instance = await createEditorFrame();
+  const storage = new Storage(localStorage);
+  const stateTransfer = embeddable?.getStateTransfer(params.history);
+  const embeddableEditorIncomingState = stateTransfer?.getIncomingEditorState();
+
+  const lensServices: LensAppServices = {
+    data,
+    storage,
+    navigation,
+    attributeService,
+    http: coreStart.http,
+    chrome: coreStart.chrome,
+    overlays: coreStart.overlays,
+    uiSettings: coreStart.uiSettings,
+    application: coreStart.application,
+    notifications: coreStart.notifications,
+    savedObjectsClient: coreStart.savedObjects.client,
```

Comment:
😆

Reply:
When people say 'code structure' this is what they mean. The pyramid is the strongest shape!